### PR TITLE
chore/style: fix Clippy lints

### PR DESCRIPTION
**Pull request by afck**

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
@afck: no appropriate reviewer found, use r? to override

File: src/message_filter.rs [hunk 1]
```diff
@@ -18,11 +18,13 @@
 #[cfg(test)]
 extern crate rand;
 
+#[cfg_attr(feature="clippy", allow(useless_attribute))]
```
Comment (Fraser999) [inline src/message_filter.rs]: I normally add a todo if I'm temporarily disabling a lint (e.g. https://github.com/Fraser999/routing/blob/leave_merge/src/message_filter.rs#L23-L24)

Comment (afck) [inline src/message_filter.rs]: Done.

**Pull request merged**